sequencer/src/l1_client.rs

```diff
@@ -129,6 +130,34 @@ impl L1Client {
             }
         }
     }
+    /// Get fee info for each `Deposit` ocurring between `prev`
+    /// and `new`. Returns `Vec<FeeInfo>`
+    pub async fn get_pending_deposits(
+        &self,
+        prev_finalized: Option<u64>,
+        new_finalized: u64,
+        address: Address,
+    ) -> Vec<FeeInfo> {
+        let prev = prev_finalized.unwrap_or(0);
+        // When called from `Header::new` during testing both will be
+        // `0`. BUG: https://github.com/foundry-rs/foundry/issues/7245
+        // We treat this as if there were no deposits.
```

Comment:
This will also frequently happen in production, as there will be many HotShot blocks produced in between each finalized ETH block. In this case returning an empty list is the correct behavior: there are in fact no new deposits.

Reply:
adjusted the comment in d85c24bb94c26ad18d1fb202f110d67f6f90de3e